core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -338,9 +348,11 @@ abstract class AbstractFetcherThread(name: String,
                     case e: KafkaStorageException =>
                       error(s"Error while processing data for partition $topicPartition", e)
                       partitionsWithError += topicPartition
-                    case e: Throwable =>
-                      throw new KafkaException(s"Error processing data for partition $topicPartition " +
-                        s"offset ${currentFetchState.fetchOffset}", e)
+                    case t: Throwable =>
+                      // drop this partition from the fetcher thread and store in a set for failed partitions
+                      error(s"Unexpected error occurred while processing data for partition $topicPartition" +
```

Comment:
nit: missing space at the end. Perhaps ".. while processing data for partition $topicPartition **at** "?